server/base/src/main/java/org/apache/accumulo/server/ServerInfo.java

```diff
@@ -176,7 +177,10 @@ public String getInstanceName() {
   }
 
   public Credentials getCredentials() {
-    return SystemCredentials.get(getInstanceID(), getSiteConfiguration());
+    if (credentials == null) {
+      credentials = SystemCredentials.get(getInstanceID(), getSiteConfiguration());
```

Comment:
This probably needs sync, volatile, or an AtomicReference to be thread safe.

This comment may be moot because of the other comment @ctubbsii  made about moving this to the constructor which would make it thread safe.